crates/router/src/routes/refunds.rs

```diff
@@ -209,3 +209,23 @@ pub async fn refunds_list(
     )
     .await
 }
+
+#[instrument(skip_all, fields(flow = ?Flow::RefundsList))]
+#[cfg(feature = "olap")]
+// #[post("/list")]
+pub async fn refunds_list_with_filters(
+    state: web::Data<AppState>,
+    req: HttpRequest,
+    payload: web::Json<api_models::refunds::RefundListRequest>,
```

Comment:
Do we need this to be a JSON payload, or can these be accepted as query params?

Also, please add documentation corresponding to the OpenAPI spec, and update the query parameters for the `refunds_list()` route handler.

https://github.com/juspay/hyperswitch/blob/9090749c661ab9ad5c756749aa947cb6f67da339/crates/router/src/routes/refunds.rs#L177-L185